aggregator_core/src/datastore.rs

```diff
@@ -559,16 +559,6 @@ impl<C: Clock> Transaction<'_, C> {
         self.run_op(self.raw_tx.query_opt(statement, params)).await
     }
 
-    /// Calling this method will force the transaction to eventually be rolled back and retried; all
-    /// datastore writes in this try will be lost. Calling this method does not interrupt or
-    /// otherwise directly affect the transaction-processing callback; the caller may wish to e.g.
-    /// use an error or some other signalling method to cause the callback to terminate.
-    ///
-    /// There is no upper limit on the number of retries a single transaction may incur.
-    pub fn retry(&self) {
-        self.retry.store(true, Ordering::Relaxed);
-    }
```

Comment:
I suppose we can revert https://github.com/divviup/janus/pull/2809 (main branch only), since now there's no way for the application to trigger unbounded retries.

Reply:
I'd like to keep it -- while I can't think of a plausible way to hit a retry loop with the current code, IMO we may as well keep the mitigation in place in case we invent a new way.